tools/tdbg/dlq_v2_service.go

```diff
@@ -60,22 +112,28 @@ func NewDLQV2Service(
 	clientFactory ClientFactory,
 	writer io.Writer,
 	prompter *Prompter,
+	taskBlobEncoder TaskBlobEncoder,
 ) *DLQV2Service {
+	encoder := newEncoder(writer)
 	return &DLQV2Service{
-		category:      category,
-		sourceCluster: sourceCluster,
-		targetCluster: targetCluster,
-		clientFactory: clientFactory,
-		writer:        writer,
-		prompter:      prompter,
-		marshaler: jsonpb.Marshaler{
-			Indent:       "  ",
-			EmitDefaults: true,
-		},
+		category:        category,
+		sourceCluster:   sourceCluster,
+		targetCluster:   targetCluster,
+		clientFactory:   clientFactory,
+		writer:          writer,
+		encoder:         encoder,
```

Comment:
Took me some time to understand why there's an encoder here and then another encoder is created in ReadMessages